refinery/data_set_manager/tasks.py

```diff
@@ -253,6 +253,7 @@ def create_dataset(investigation_uuid, username, identifier=None, title=None,
     if public:
         public_group = ExtendedGroup.objects.public_group()
         dataset.share(public_group)
+    annotate_nodes(investigation_uuid)
```

Comment:
Relocate annotation after dataset is created

Reply:
If we need to `annotate_nodes()` whenever a dataset is created, then I think we could move this call to the post_save signal handler within an if like:

```
if kwargs["created"]:
    investigation = instance.get_investigation()
    if investigation is not None:
        annotate_nodes(investigation.uuid)
```

Reply:
Talked in person, confusion about where the investigation is set (post creation). All set.

Reply:
Overriding `save()` is generally recommended since neither `pre_save` nor `post_save` are called during bulk operations anyway and it allows keeping model update code in one place.
https://docs.djangoproject.com/en/1.8/topics/db/models/#overriding-predefined-model-methods